File: src/main/java/com/christophsens/cvenhancer/utils/IsoUtils.java
package com.christophsens.cvenhancer.utils;

import java.util.Locale;
import java.util.Set;

public final class IsoUtils {
    private static final Set<String> ISO_LANGUAGES = Set.of(Locale.getISOLanguages());
    private static final Set<String> ISO_COUNTRIES = Set.of(Locale.getISOCountries());

    private IsoUtils() {
    }
    private static boolean isValidISOLanguage(String s) {
        return ISO_LANGUAGES.contains(s);
    }
    private static boolean isValidISOCountry(String s) {
        return ISO_COUNTRIES.contains(s);
    }
    public static boolean isValidIsoLocale(String languageCode, String countryCode) {
        return isValidISOLanguage(languageCode) && isValidISOCountry(countryCode); 
    }
}

File: src/main/kotlin/com/christophsens/cvenhancer/utils/IsoUtils.kt
package com.christophsens.cvenhancer.utils

import java.util.Locale

object IsoUtils {

    private val isoLanguages: Set<String> = Locale.getISOLanguages().toSet()
    private val isoCountries: Set<String> = Locale.getISOCountries().toSet()

    private fun isValidIsoLanguage(value: String): Boolean {
        return isoLanguages.contains(value)
    }

    private fun isValidIsoCountry(value: String): Boolean {
        return isoCountries.contains(value)
    }

    fun isValidIsoLocale(languageCode: String, countryCode: String): Boolean {
        return isValidIsoLanguage(languageCode) && isValidIsoCountry(countryCode)
    }
}